components/eamxx/src/share/atm_process/atmosphere_process.hpp

```diff
@@ -276,6 +280,13 @@ class AtmosphereProcess : public ekat::enable_shared_from_this<AtmosphereProcess
   // For BFB tracking in production simulations.
   void print_fast_global_state_hash(const std::string& label) const;
 
+  // Set IOP object
+  virtual void set_intensive_observation_period(const iop_ptr& iop) {
+    m_iop = iop;
+  }
+
+  const iop_ptr& get_intensive_observation_perdiod() { return m_iop; }
```

Comment:
I would just make `m_iop` protected, so that derived classes can access it without the lengthy `get_intensive_operation_period()` call. After all, if they can set and get, they should just have access to the member.